migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,173 @@
+    display_order                       smallint                                        ,
+    active_ind                          boolean DEFAULT true                    NOT NULL,
+    create_user                         character varying(60)                   NOT NULL,
+    create_timestamp                    timestamp with time zone DEFAULT now()  NOT NULL,
+    update_user                         character varying(60)                   NOT NULL,
+    update_timestamp                    timestamp with time zone DEFAULT now()  NOT NULL
+);
+
+COMMENT ON TABLE mine_report_submission_status_code is 'All the possible and available status codes for mine reports.';
+ALTER TABLE mine_report_submission_status_code OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_definition (
+    mine_report_definition_id            serial                                  PRIMARY KEY, 
+    mine_report_definition_guid          uuid UNIQUE                             NOT NULL   ,
+    report_name                          varchar(50)                             NOT NULL   ,
+    description                          varchar(300)                            NOT NULL   ,
+    due_date_period_months               integer                                 NOT NULL   ,
+    mine_report_due_date_type            character varying(3)                    NOT NULL   ,
+    active_ind                           boolean DEFAULT true                    NOT NULL   ,
+    create_user                          character varying(60)                   NOT NULL   ,
+    create_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+    update_user                          character varying(60)                   NOT NULL   ,
+    update_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+
+    FOREIGN KEY (mine_report_due_date_type) REFERENCES mine_report_due_date_type(mine_report_due_date_type) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_definition is 'Definitions of code required reports, due dates and articles of the code that apply to them.';
+ALTER TABLE mine_report_definition OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report
+(
+    mine_report_id               serial                                   NOT NULL PRIMARY KEY,
+    mine_report_guid             uuid UNIQUE                              NOT NULL            ,
+    mine_report_definition_id    integer                                  NOT NULL            ,
+    mine_guid                    uuid                                     NOT NULL            ,
+    permit_id                    integer                                                      ,
+    due_date                     timestamp                                NOT NULL            ,
+    received_date                timestamp                                                    ,
+    submission_year              smallint                                                     ,
```

Comment:
Most submissions are done on fiscal year are they not? Should this be character so that "18/19" is accepted? Or does "19" imply "18/19"?

Reply:
We've talked about how this will be tricky because it's really a compliance _period,_ which may be a calendar year, or a rolling 5 year period. But I think most often the compliance year is a calendar year (eg TSF reports, though due on Mar 31 for some random reason, apply to the previous calendar year - if an event happens on Jan 30, it will be covered in the TSF report due Mar 31 of the following year.)